lib/matplotlib/tests/test_ticker.py

```diff
@@ -1594,6 +1594,79 @@ def test_engformatter_usetex_useMathText():
         assert x_tick_label_text == ['$0$', '$500$', '$1$ k']
 
 
+@pytest.mark.parametrize(
+    'oom_center, oom_noise, oom_center_desired, oom_noise_desired', [
+        (11, 1, 9, 0),
+        (13, 7, 12, 6),
+        (1, -2, 0, -3),
+        (3, -2, 3, -3),
+        (5, -3, 3, -3),
+        (2, -3, 0, -3),
+        # The following sets of parameters demonstrates that when oom_center-1
+        # and oom_noise-2 equal a standard 3*N oom, we get that
+        # oom_noise_desired < oom_noise
+        (10, 2, 9, 3),
+        (1, -7, 0, -6),
+        (2, -4, 0, -3),
+        (1, -4, 0, -3),
+        # Tests where oom_center <= oom_noise, those are probably covered by the
+        # part where formatter.offset != 0
+        (4,  4, 0, 3),
+        (1,  4, 0, 3),
+        (1,  3, 0, 3),
+        (1,  2, 0, 0),
+        (1,  1, 0, 0),
+    ]
+)
+def test_engformatter_offset_oom(
+    oom_center,
+    oom_noise,
+    oom_center_desired,
+    oom_noise_desired
+):
+    UNIT = "eV"
+    # Doesn't really matter here, but should be of order of magnitude ~= 1
+    r = range(-5, 7)
+    fig, ax = plt.subplots()
+    # Use some random ugly number
+    data_offset = 2.7149*10**oom_center
+    ydata = data_offset + np.array(r, dtype=float)*10**oom_noise
+    ax.plot(ydata)
+    formatter = mticker.EngFormatter(useOffset=True, unit=UNIT)
+    # So that offset strings will always have the same size
+    formatter.ENG_PREFIXES[0] = "_"
+    ax.yaxis.set_major_formatter(formatter)
+    fig.canvas.draw()
+    offsetGot = formatter.get_offset()
+    ticksGot = [labl.get_text() for labl in ax.get_yticklabels()]
+    # Predicting whether offset should be 0 or not is essentially testing
+    # ScalarFormatter._compute_offset . This function is pretty complex and it
+    # would be nice to test it, but this is out of scope for this test which
+    # only makes sure that offset text and the ticks gets the correct unit
+    # prefixes and the ticks.
+    if formatter.offset:
+        prefix_noise_got = offsetGot[2]
+        prefix_noise_desired = formatter.ENG_PREFIXES[oom_noise_desired]
+        prefix_center_got = offsetGot[-1-len(UNIT)]
+        prefix_center_desired = formatter.ENG_PREFIXES[oom_center_desired]
+        assert prefix_noise_desired == prefix_noise_got
+        assert prefix_center_desired == prefix_center_got
```

Comment:
I'd just inline these; they aren't used elsewhere:
```suggestion
        assert formatter.ENG_PREFIXES[oom_noise_desired] == offsetGot[2]
        assert formatter.ENG_PREFIXES[oom_center_desired] == offsetGot[-1-len(UNIT)]
```

Reply:
> I'd just inline these; they aren't used elsewhere:

I thought about this too, but while debugging the tests with `pytest --showlocals`, I hope it is not a deal breaker to keep those, as I plan to maintain these tests and this feature in the future. I added a comment explaining that.

Reply:
Same here.